dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/providers/saml/SamlFailedLoginPage.java

```diff
@@ -0,0 +1,48 @@
+package edu.harvard.iq.dataverse.authorization.providers.saml;
+
+import edu.harvard.iq.dataverse.authorization.SamlLoginIssue;
+import edu.harvard.iq.dataverse.common.BundleUtil;
+import edu.harvard.iq.dataverse.util.JsfHelper;
+import org.omnifaces.cdi.ViewScoped;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.faces.context.FacesContext;
+import javax.inject.Named;
+import javax.servlet.http.HttpSession;
+import java.io.Serializable;
+
+@ViewScoped
+@Named("FailedLogin")
+public class SamlFailedLoginPage implements Serializable {
+
+    private static final Logger logger = LoggerFactory.getLogger(SamlFailedLoginPage.class);
+
+    private SamlLoginIssue samlLoginIssue;
+
+    // -------------------- GETTERS --------------------
+
+    public SamlLoginIssue getSamlLoginIssue() {
+        return samlLoginIssue;
+    }
+
+    // -------------------- LOGIC --------------------
+
+    public void init() {
+        HttpSession session = (HttpSession) FacesContext.getCurrentInstance().getExternalContext().getSession(false);
+        SamlLoginIssue loginIssue = session != null
+                ? (SamlLoginIssue) session.getAttribute(SamlAuthenticationServlet.SAML_LOGIN_ISSUE_SESSION_PARAM) : null;
+        if (loginIssue != null) {
+            samlLoginIssue = loginIssue;
+            session.removeAttribute(SamlAuthenticationServlet.SAML_LOGIN_ISSUE_SESSION_PARAM);
+        } else {
+            logger.warn("Cannot find session or original login issue. Using AUTHENTICATION_ERROR.");
+            samlLoginIssue = new SamlLoginIssue(SamlLoginIssue.Type.AUTHENTICATION_ERROR);
+        }
+        String errorMessage = BundleUtil.getStringFromBundle("failed.login.cause." + samlLoginIssue.type.toString().toLowerCase().replace("_", "."));
```

Comment:
To be honest I would rather have simple `samlLoginIssue.type.toString()` `samlLoginIssue.type.toString().toLowerCase().replace("_", ".")`. That way when I would be looking up where in the code is showing up some message I wouldn't be wasting my time by searching by the full key (for example. `failed.login.cause.invalid.data`) but at first try I would search the app for the string `failed.login.cause.` Because I can already see that there is something unusual with the full key name. Maybe it is just a personal taste. What do you think

Reply:
I wanted to have `INVALID_DATA` in enum and `invalid.data` in message name in order to have consistent format everywhere. Now I think it would be better to have the message name added explicitly to enum – I'd try to fix that in future.